### PR TITLE
fix: correct detection of Spark server

### LINKED ISSUE #265: A Spring Boot app that's configured to use Jetty fails to generate request recordings
With this in a Spring Boot app's `pom.xml`

```xml
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-web</artifactId>
      <exclusions>
        <exclusion>
          <groupId>org.springframework.boot</groupId>
          <artifactId>spring-boot-starter-tomcat</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-jetty</artifactId>
    </dependency>
```
the `mvn spring-boot:run` will start an embedded Jetty server to host the app.

The Java agent incorrectly detects this as a Spark server and causes request recording to fail.

### PR DESCRIPTION
**Pull request by apotterri**
When instrumenting a Jetty server, only add a Spark handler if the app is really a Spark app. A mistake in the code meant that it added the handler whenever running in a Jetty server. This would cause request recording to fail (and maybe cause other problems, as well).

Fixes #265 .

### CODE REVIEW

File: agent/src/main/java/com/appland/appmap/process/hooks/Spark.java [hunk 1]
```diff
@@ -145,11 +145,16 @@ public static void setHandler(Event event, Object receiver, Object[] args) {
...
+        boolean match = Arrays.stream(hl.getHandlers())
+            .anyMatch(h -> h.getClass().getName().equals(jettyHandlerClass));
         if (!match) {
           return;
         }
       }
+
+      // If it's not either the Spark JettyHandler or a HandlerList, we're not running in a Spark
+      // server.
+      return;
```
Comment (dividedmind) [inline agent/src/main/java/com/appland/appmap/process/hooks/Spark.java:157]: Won't this always return for a HandlerList regardless whether it contains a JettyHandler? 
Perhaps this logic would be clearer in a separate utility function.

**Review by dividedmind — COMMENTED** [review]
I'm not sure this change does what it should.

**Comment by apotterri** [comment]
Hmmm, I think you're right. Given that, it's weird that the Spark tests pass. I'll investigate why that is.

**Review by dividedmind — APPROVED** [review]
Thanks, now it clearly does what it says on the tin!

**Pull request merged**